ee/clickhouse/queries/test/test_lifecycle.py

```diff
@@ -70,13 +76,53 @@ def test_test_account_filters_with_groups(self):
             self.team,
         )
 
-        self.assertEqual(sorted(res["status"] for res in result), ["dormant", "new", "resurrecting", "returning"])
-        for res in result:
-            if res["status"] == "dormant":
-                self.assertEqual(res["data"], [0, -1, 0, 0, -1, 0, 0, 0])
-            elif res["status"] == "returning":
-                self.assertEqual(res["data"], [0, 0, 0, 0, 0, 0, 0, 0])
-            elif res["status"] == "resurrecting":
-                self.assertEqual(res["data"], [1, 0, 0, 1, 0, 0, 0, 0])
-            elif res["status"] == "new":
-                self.assertEqual(res["data"], [0, 0, 0, 0, 0, 0, 0, 0])
+        self.assertLifecycleResults(
+            result,
+            [
+                {"status": "dormant", "data": [0, -1, 0, 0, -1, 0, 0, 0]},
+                {"status": "new", "data": [0, 0, 0, 0, 0, 0, 0, 0]},
+                {"status": "resurrecting", "data": [1, 0, 0, 1, 0, 0, 0, 0]},
+                {"status": "returning", "data": [0, 0, 0, 0, 0, 0, 0, 0]},
+            ],
+        )
+
+    @snapshot_clickhouse_queries
+    def test_lifecycle_edge_cases(self):
+        # This test tests behavior when created_at is different from first matching event and dormant/resurrecting/returning logic
+        with freeze_time("2020-01-11T12:00:00Z"):
+            Person.objects.create(distinct_ids=["person1"], team_id=self.team.pk)
+
+        journeys_for(
+            {
+                "person1": [
+                    {"event": "$pageview", "timestamp": datetime(2020, 1, 12, 12),},
```

Comment:
Are all these events needed? If we are just concerned with the NEW case mentioned above, then would one event be a sufficient for a minimal test case?